### ISSUE TITLE
logstash json filter throws exception and stops processing after valid but bad json

### ISSUE THREAD

**Issue by llchen223** [elastic/logstash#8882]
I'm seeing Logstash 6.1.1 stop processing events after exception on valid json that works on Logstash 1.4.2.  Expected result is for logstash to skip and tag the record with parse failure:

- Version: Logstash 6.1.1
- Operating System: Ubuntu 17.10
- Config File (if you have sensitive info, please remove it):
```
input {
  stdin {
  }
}

filter {
  json {
    source => "message"
  }
}
  
output {
  stdout {codec => json} 
}
```

- Sample Data:
```
{"foo":"bar", "":0}
```
- Steps to Reproduce:
Input the sample data into stdin.  Logstash 6.1.1 will throw exception and stop processing new records.

Here's the exception message:
```
[2017-12-22T14:19:25,314][ERROR][logstash.pipeline        ] Exception in pipelineworker, the pipeline stopped processing new events, please check your filter configuration and restart Logstash. {:pipeline_id=>"main", "exception"=>"-1", "backtrace"=>["java.util.ArrayList.elementData(ArrayList.java:422)", "java.util.ArrayList.remove(ArrayList.java:499)", "org.logstash.FieldReference.parse(FieldReference.java:159)", "org.logstash.FieldReference.parseToCache(FieldReference.java:146)", "org.logstash.FieldReference.from(FieldReference.java:78)", "org.logstash.ext.JrubyEventExtLibrary$RubyEvent.ruby_set_field(JrubyEventExtLibrary.java:88)",
...
Exception in thread "Ruby-0-Thread-8@[main]>worker0: /home/lchen/Downloads/logstash-6.1.1/logstash-core/lib/logstash/pipeline.rb:392" java.lang.ArrayIndexOutOfBoundsException: -1
	at java.util.ArrayList.elementData(ArrayList.java:422)
	at java.util.ArrayList.remove(ArrayList.java:499)
	at org.logstash.FieldReference.parse(FieldReference.java:159)
	at org.logstash.FieldReference.parseToCache(FieldReference.java:146)
	at org.logstash.FieldReference.from(FieldReference.java:78)
	at org.logstash.ext.JrubyEventExtLibrary$RubyEvent.ruby_set_field(JrubyEventExtLibrary.java:88)
	at org.logstash.ext.JrubyEventExtLibrary$RubyEvent$INVOKER$i$2$0$ruby_set_field.call(JrubyEventExtLibrary$RubyEvent$INVOKER$i$2$0$ruby_set_field.gen)
	at org.jruby.runtime.callsite.CachingCallSite.cacheAndCall(CachingCallSite.java:358)
	at org.jruby.runtime.callsite.CachingCallSite.call(CachingCallSite.java:195)
	at org.jruby.ir.interpreter.InterpreterEngine.processCall(InterpreterEngine.java:323)
	at org.jruby.ir.interpreter.StartupInterpreterEngine.interpret(StartupInterpreterEngine.java:73)
```

This is output from Logstash 1.4.2:
```
{"message":"{\"foo\":\"bar\", \"\":0}","@version":"1","@timestamp":"2017-12-22T22:28:25.547Z","host":"ubuntu","foo":"bar","":0}
```

**Comment by btungut** [comment]
Same +1
Key-value object with empty key should not be cause this.

https://github.com/elastic/logstash/issues/8885

**Comment by jordansissel** [comment]
Duplicate of https://github.com/elastic/logstash/issues/8823

Fixed in https://github.com/elastic/logstash/pull/8824; first release with fix is Logstash 6.1.1

**Comment by jsvd** [comment]
This fix is actually in 6.1.2
